### PR TITLE
Update comments in http.h

### PR DESCRIPTION
**Pull request by ev-ermakov**
fixes #

Note that the above format should be used in your git commit comments.
You agree that by submitting a PR, you have read and agreed to our
contributing guidelines.

### CODE REVIEW

**Comment by ev-ermakov** [comment]
```c
// The callback should return 0 in most circumstances; ...
```
Oh, maybe I misunderstood this comment. The callback returns void itself.

File: include/nng/supplemental/http/http.h [hunk 1]
```diff
@@ -305,10 +305,6 @@ typedef struct nng_http_handler nng_http_handler;
 // Upon completion, the handler should store an nng_http_res * as the
 // first output using nng_aio_set_output.  If it does not do so, or supplies
 // NULL, then it must send a response itself.
-//
-// The callback should return 0 in most circumstances; if it returns anything
-// other than 0 then the connection is terminated (after possibly sending
-// a 500 error response to the client.)
```
Comment (gdamore) [inline include/nng/supplemental/http/http.h]: Why did you remove this comment.  I think it's accurate, no?

Comment (gdamore) [reply include/nng/supplemental/http/http.h]: I see the problem -- the issue is the word "return".
This should have have been reworded as:

"The callback should complete with a result of 0 most circumstances.  If it completes with an error,
then the connection is terminated, after possibly sending a 500 error response to the client"

Comment (gdamore) [reply include/nng/supplemental/http/http.h]: The critical information we need to convey is that a non-successful completion of the callback closes the connection (maybe returning a 500 error code as well.)

Comment (ev-ermakov) [reply include/nng/supplemental/http/http.h]: Agreed! I updated the pull request with the suggested changes.
